CONTRIBUTING.md

```diff
@@ -0,0 +1,121 @@
+
+While issues are a great way to discuss problems, bugs and new features, a direct proposal via a pull request can sometimes say more than a thousand words. So be bold and contribute to the code as described in the [next section](#contribute-to-source-code)!
+
+In case you require a more private communication, you can reach us via [connaisseur@securesystems.de](mailto:connaisseur@securesystems.de).
+
+## Contribute to Source Code
+The following steps will help you make code contributions to Connaisseur and ensure good code quality and workflow. This includes the following steps:
+1. **Setup your environment**: Setup up your local environment to best interact with the code. Further information is given in the subsection below.
+2. **Make atomic changes**: Changes should be atomic, i.e. each pull request should only fix one issue or implement one feature. As such, each pull request should ideally only contain a single commit with an concise commit message.
+3. **Test your changes**: Test any changes locally for code quality and functionality and add new tests for any additional code. How to test is described in the subsection below.
+4. **Create semantic, conventional and signed commits**: Any commits should follow a simple semantic convention to help structure the work on Connaisseur. The convention is described in the corresponding subsection. For security reasons and as the project is about signing code, we require signed commits.
+5. **Create Pull Request**: We consider code review a critical measure for code quality and security. Therefore, a pull request (PR) should be created for each contribution. It will be reviewed by one of the main contributors and potential improvements may be discussed within the PR. At least one approval is required for merging changes into the `master` branch.
+
+### Setup the Environment
+To start contributing, you will need to setup your local environment. First step is to get the source code by cloning this repository:
+```bash
+git clone git@github.com:sse-secure-systems/connaisseur.git
+```
+In order to review the effects of your changes, you should create your local Kubernetes cluster and install Connaisseur. This is described in the [setup guide](setup/README.md). A simple starting point may be a Minikube cluster with e.g. a [Docker Hub](https://hub.docker.com/) repository for maintaining your test images and trust data.
+
+### Test Changes
+Test are important to ensure code quality, functionality and security and therefore we aim to keep the code coverage high. We are running several automated tests in the [CI pipeline](https://github.com/sse-secure-systems/connaisseur/blob/master/.github/workflows/cicd.yaml). Application code is tested via [pytest](https://docs.pytest.org/). When making changes to the application code, please directly provide tests for your changes.
+
+To enforce linting and reduce review effort, you can use [black](https://pypi.org/project/black/) for linting. You can install via:
```

Comment:
We use pylint for linting, not black. Black is just used for autoformatting the code, which will bring the code into a shape pylint will accept. But missused variables or unnessecary imports are only detected by pylint and not fixed by black's formatting.